GenericTensor/Functions/EchelonForm.cs

```diff
@@ -223,18 +242,50 @@ internal static void InnerGaussianEliminationSimple(GenTensor<T, TWrapper> t, in
 
         public static GenTensor<T, TWrapper> RowEchelonFormSimple(GenTensor<T, TWrapper> t)
         {
-            #if ALLOW_EXCEPTIONS
+#if ALLOW_EXCEPTIONS
             if (!t.IsMatrix)
                 throw new InvalidShapeException("this should be matrix");
-            #endif
+#endif
             var res = t.Copy(copyElements: false);
             InnerGaussianEliminationSimpleDiscardSwapCount(res, 0);
             return res;
         }
 
+        public static (GenTensor<T, TWrapper>, int[]) RowEchelonFormPermuteSimple(GenTensor<T, TWrapper> t)
+        {
+#if ALLOW_EXCEPTIONS
+            if (!t.IsMatrix)
+                throw new InvalidShapeException("this should be matrix");
+#endif
+            var res = t.Copy(copyElements: false);
+            var permute = new int[t.Shape[0]];
+            for (var i = 0; i < permute.Length; i++) permute[i] = i + 1;
```

Comment:
Inconsistent style of loops with https://github.com/asc-community/GenericTensor/pull/26/files#diff-5707b7048aaf015276f94f17a53dddd0af7981df5be41bf29a8ca9376a878997R276